### ISSUE TITLE
AddFrameListener not calling the S4FrameProc

### ISSUE THREAD

**Issue by WizzardMaker** [nyfrk/S4ModApi#4]
I add a S4FrameProc with the AddFrameListener function and it returns a non zero value, but the function never gets called.

I tried calling AddFrameListener in DLL_PROCESS_ATTACH and mid game, but both don't work.

This is using the latest release build of the S4ModApi

**Comment by nyfrk** [comment]
The FrameListener works for all (unmodded) known Editions and Versions of the Game (History and Gold Edition, including the versions without AddOn). It should also work for the free Demo Version. 

Make sure that there is no other plugin in your plugins directory that statically links against the S4ModApi. I know some plugins that statically link:

* Vicitien's Dark Tribe Plugin
* nyfrk's Unlimited Selection Plugin

The S4ModApi was never intended to work when statically linked. Maybe I will add support for statically linking in the future. But that's not one of my top priorities.

You may ask the authors of the conflicting plugin to release the source code so you can create a dynamically linked fork of their plugin (statically linking plugins must be released under GPL, since the S4ModApi is licensed under LGPL).

If I happen to be the author of the conflicting plugin, then I will update that in the future. As of now, the API / ABI is not stable enough to create a release of the S4ModApi.

In conclusion: You must dynamically link all plugins to the same S4ModApi.dll.
